myia/compile/backends/python/code_generator.py

```diff
@@ -331,7 +339,14 @@ def _str_make_dict(self, *inputs):
 
     def _str_apply(self, fn, args, kwargs):
         """Formatter for apply `apply`, which seems to represent a call with args and kwargs."""
-        return f"{self._label(fn)}(*{self._rvalue(args)}, **{self._rvalue(kwargs)})"
+        params = []
+        l, _ = args.args()
+        for e in l:
+            params.append(f"*{self._rvalue(e)}")
+        l, _ = kwargs.args()
+        for e in l:
+            params.append(f"**{self._rvalue(e)}")
+        return f"{self._label(fn)}({', '.join(params)})"
```

Comment:
Requiring `apply` to have two `concat` as arguments (and not checking that they are `concat`) is somewhat fragile. I would find it less ad hoc to have `concat` actually do something and generate `f(*tuple_concat(a, b), **dict_concat(c, d))`, or `f(*(*a, *b), **{**c, **d})`. Then `apply` can take arbitrary expressions and still work.

Reply:
I'm ok with that but then how is that really different from star and dstar?